### PR TITLE
[docs] Split Jobs to Asset Jobs and Op Jobs

### PR DESCRIPTION
**Pull request by yuhan**
## Summary & Motivation
this is a follow up to the docs left nav updates. tracking doc: https://www.notion.so/dagster/docs-Split-Jobs-to-Asset-Jobs-and-Op-Jobs-87ba2207cf074e01850ceef7cbc14b49

After landing the left nav changes, we felt we were getting a lil ahead of ourselves by marking all jobs as advanced. Instead of doing so, this PR splits Jobs into "Asset jobs" and "Op jobs", then treat "Asset jobs" as an essential concept under "Assets and keep "Op jobs" in "Advanced":

Screenshot 2023-08-28 at 7 32 09 PM

There's no significant content changes. Mostly just split the page into two and cross-reference to point users to the right place.

Worth noting that all "Job execution" and "Metadata & tags" don't talk about assets, so it might be worth creating more sections in asset-jobs to explain asset job execution and metadata like using asset-centric UI.

### CODE REVIEW

**Comment by yuhan** [comment]
Current dependencies on/for this PR:
* master
  * **PR #16156** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/16156" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/16156?utm_source=stack-comment).

File: docs/content/concepts/assets/asset-jobs.mdx
Comment (jamiedemaria) [inline docs/content/concepts/assets/asset-jobs.mdx:186]: does this guide have examples for asset jobs? if it's just op jobs, then we should remove the link

**Comment by yuhan** [comment]
re: config, yes we'll need to write new content for configuring asset jobs. but the commenting was an oversight - that was supposed to be removed.

File: docs/content/_navigation.json
Comment (erinkcochran87) [inline docs/content/_navigation.json:82]:
```suggestion
            "title": "Asset jobs",
```

No cap needed

File: docs/content/concepts/assets/asset-jobs.mdx
Comment (erinkcochran87) [inline docs/content/concepts/assets/asset-jobs.mdx:16]: I suggest we move this above the intro paragraph. That way it's the first thing someone sees when they come here, which is great if ops is what they're looking for.

File: docs/content/concepts/assets/asset-jobs.mdx
Comment (erinkcochran87) [inline docs/content/concepts/assets/asset-jobs.mdx:19]:
```suggestion
  Software-defined Assets? Visit the{" "}
```

File: docs/content/concepts/assets/asset-jobs.mdx
Comment (erinkcochran87) [inline docs/content/concepts/assets/asset-jobs.mdx:61]:
```suggestion
Unlike jobs created using the <PyObject object="job" decorator /> decorator where you explicitly define the dependencies when you create the job, the topology of an asset-based job is based on the [assets](/concepts/assets/software-defined-assets) and their dependencies.
```

Missing `decorator` argument

Comment (erinkcochran87) [reply docs/content/concepts/assets/asset-jobs.mdx:61]: +1

File: docs/content/concepts/assets/asset-jobs.mdx
Comment (erinkcochran87) [inline docs/content/concepts/assets/asset-jobs.mdx:155]:
```suggestion
You make jobs available to the Dagster UI, GraphQL, and the command line by including them in <PyObject object="Definitions"/> object at the top level of a Python module or file. The tool loads that module as a code location. If you include schedules or sensors, the code location will automatically include jobs that those schedules or sensors target.
```

Comment (erinkcochran87) [reply docs/content/concepts/assets/asset-jobs.mdx:155]: Consider adding a link to the Code location docs in here somewhere: `/concepts/code-locations`

File: docs/content/concepts/assets/asset-jobs.mdx
Comment (erinkcochran87) [reply docs/content/concepts/assets/asset-jobs.mdx:186]: It doesn't, currently. Agreed we should remove this until it does.

File: docs/content/concepts/ops-jobs-graphs/jobs.mdx
Comment (erinkcochran87) [inline docs/content/concepts/ops-jobs-graphs/jobs.mdx:12]: Including this in here feels really weird to me, since this page is supposed to be about op jobs. In that case, why would the description include assets?

File: docs/content/concepts/ops-jobs-graphs/jobs.mdx
Comment (erinkcochran87) [inline docs/content/concepts/ops-jobs-graphs/jobs.mdx:13]: Same here, re: including assets feels weird

File: docs/content/concepts/ops-jobs-graphs/jobs.mdx
Comment (erinkcochran87) [inline docs/content/concepts/ops-jobs-graphs/jobs.mdx:35]: Again, this feels weird to me. Maybe we should discuss since it keeps coming up?

File: docs/content/concepts/ops-jobs-graphs/jobs.mdx
Comment (erinkcochran87) [inline docs/content/concepts/ops-jobs-graphs/jobs.mdx:33]: Why wouldn't this just be "Creating op jobs"?

File: docs/content/concepts/ops-jobs-graphs/jobs.mdx
Comment (erinkcochran87) [inline docs/content/concepts/ops-jobs-graphs/jobs.mdx:40]: I suggest moving this up to the top of the page, right after the page title, so it mirrors what you have for the asset jobs page.

File: docs/content/concepts/ops-jobs-graphs/jobs.mdx
Comment (erinkcochran87) [inline docs/content/concepts/ops-jobs-graphs/jobs.mdx:45]: I can't comment on some of the things I see since they're not technically your changes, but if you don't mind addressing these:

- [ ] - Line 49 `job` is missing the decorator. Could you please add it? `<PyObject object="job" decorator />`
- [ ] - The `Using the @job decorator` and `From a graph` headings should be H3s, not H4
- [ ] - Line 125 - `software-defined assets` should be `Software-defined Assets`
- [ ] - Line 135 - This paragraph says `the UI`, but as it's now the first mention of the UI on this page, it should be `the Dagster UI`
- [ ] - Line 167 - Should `op-based jobs` now be `op jobs`?

**Comment by erinkcochran87** [comment]
@jamiedemaria Back to you for another round' of reviewing! 🎉 

I realized that the [**Metadata & tags** page](https://docs.dagster.io/concepts/ops-jobs-graphs/metadata-tags) may also need some updates, but I'm not sure what, if anything, is specific to either ops or assets. There may also be other places in the docs where clarification is needed - I'm not sure if this is as straightforward as just splitting up the Jobs docs, since op and asset jobs are set up differently.

**Review by jamiedemaria — CHANGES_REQUESTED** [review]
nice! noticed two small things but otherwise this looks good to me 

For the metadata and tags section - I think both of those pages should be probably reworked to include asset content, and in doing that we'll get a better idea if it should be split up into op and asset specific pages

File: docs/content/concepts/ops-jobs-graphs/jobs.mdx
Comment (erinkcochran87) [reply docs/content/concepts/ops-jobs-graphs/jobs.mdx:170]: Absolutely - good catch!

Comment (erinkcochran87) [reply docs/content/concepts/ops-jobs-graphs/jobs.mdx:170]: I updated the link, but looking at the partitions page, it strikes me as super weird that the page title is `Partitioned assets and jobs` and then there's this big section with `Partitioned non-asset jobs` as the heading. Given the page title, I wouldn't expect to find anything about ops/non-assets here.

Comment (erinkcochran87) [reply docs/content/concepts/ops-jobs-graphs/jobs.mdx]: The intro is now weird too - let's discuss this when we chat on Wednesday

Comment (jamiedemaria) [reply docs/content/concepts/ops-jobs-graphs/jobs.mdx]: hmmm yeah all good points

Comment (erinkcochran87) [reply docs/content/concepts/ops-jobs-graphs/jobs.mdx]: Coming back to this, we should wait until #16511 is merged to update this link.

File: docs/content/concepts/assets/asset-jobs.mdx
Comment (erinkcochran87) [reply docs/content/concepts/assets/asset-jobs.mdx]: Updated!

**Comment by erinkcochran87** [comment]
@jamiedemaria @yuhan Back to you fine folks. I opted to adopt a similar approach to the split as I did for partitions (#16511) and make `concepts/jobs` into a concept/category page. The copy in there may need some work.

File: docs/content/concepts/ops-jobs-graphs/jobs.mdx
Comment (yuhan) [inline docs/content/concepts/ops-jobs-graphs/jobs.mdx:2]: > > make `concepts/jobs` into a concept/category page

i don't see the `concepts/jobs` in the changes. is this the page you referred as "concepts/jobs"? where do you envision it to live going forward?

File: docs/content/concepts/ops-jobs-graphs/jobs.mdx
Comment (yuhan) [inline docs/content/concepts/ops-jobs-graphs/jobs.mdx:37]: similar feedback as mentioned [here](https://github.com/dagster-io/dagster/pull/16511#discussion_r1337802711). could we merge this into the "Uses" section?

**Review by yuhan — COMMENTED** [review]
i like the direction the "Jobs" page is going (similar to the Partitions page).

File: docs/content/concepts/ops-jobs-graphs/jobs.mdx
Comment (erinkcochran87) [reply docs/content/concepts/ops-jobs-graphs/jobs.mdx:2]: Yes - sorry that wasn't clear.

File: docs/content/concepts/ops-jobs-graphs/jobs.mdx
Comment (erinkcochran87) [reply docs/content/concepts/ops-jobs-graphs/jobs.mdx:37]: Done!

File: docs/content/concepts/ops-jobs-graphs/jobs.mdx
Comment (erinkcochran87) [reply docs/content/concepts/ops-jobs-graphs/jobs.mdx:2]: Yes - sorry that wasn't clear. I'm trying to figure out where to put this in the sidenav. Here's how things are currently structured:

- `Concepts > Assets > Asset jobs`
- `Concepts > Advanced > Ops > Op jobs`

Normally I'd say `Concepts > Jobs`, but we removed that to de-emphasize the concept. We could do this, maybe?:

```
- Concepts
  - Advanced
    - Jobs (/concepts/jobs)
      - Asset jobs (/concepts/jobs/asset-jobs)
      - Op jobs (/concepts/jobs/op-jobs)
```

Comment (yuhan) [reply docs/content/concepts/ops-jobs-graphs/jobs.mdx:2]: I wonder if it'd make sense to have this for now:
```
- Concepts
  - Assets
        - Asset jobs (/concepts/assets/asset-jobs)
  - Advanced
    - Jobs (/concepts/jobs)
      - Asset jobs (/concepts/assets/asset-jobs)
      - Op jobs (/concepts/jobs/op-jobs)
```
which makes this solution a stopgap while we will be improving the core api.

and then, later this year, once we get rid of asset jobs on the critical path for assets, we can switch to:
```
- Concepts
  - Assets
  - Advanced
    - Jobs (/concepts/jobs)
      - Asset jobs (/concepts/jobs/asset-jobs) # move it down to an advanced concept
      - Op jobs (/concepts/jobs/op-jobs)
```

Comment (erinkcochran87) [reply docs/content/concepts/ops-jobs-graphs/jobs.mdx:2]: I think that's fine - it's kinda weird but I'd be surprised if anyone was confused by it.

Comment (erinkcochran87) [reply docs/content/concepts/ops-jobs-graphs/jobs.mdx:2]: I also moved `Job execution` and `Metadata & tags` under this heading, as it felt weird to leave them hanging around below it. So now it's:

```
- Concepts
  - Assets
      - Asset jobs (/concepts/assets/asset-jobs)
  - Advanced
    - Jobs (/concepts/jobs)
      - Asset jobs (/concepts/assets/asset-jobs) # links to the essential page
      - Op jobs (/concepts/jobs/op-jobs)
      - Job execution
      - Job metadata & tags
```

Comment (yuhan) [reply docs/content/concepts/ops-jobs-graphs/jobs.mdx:2]: this looks good to me! waiting on the deployment and do one last pass before stamping

File: docs/content/_navigation.json
Comment (yuhan) [inline docs/content/_navigation.json:239]:
```suggestion
                "path": "/concepts/ops-jobs-graphs/op-jobs"
```
wrong link caused 404

File: docs/content/concepts/assets/asset-jobs.mdx
Comment (yuhan) [inline docs/content/concepts/assets/asset-jobs.mdx:15]: now that we have a kinda overarching jobs page, should we also mention that page here for further info?

**Review by yuhan — COMMENTED** [review]
looks great! left one last nit - take it or leave it. (i can't approve my own pr lol. so feel free to approve it yourself and land it)

File: docs/content/_navigation.json
Comment (erinkcochran87) [reply docs/content/_navigation.json]: Fixed!

File: docs/content/concepts/assets/asset-jobs.mdx
Comment (erinkcochran87) [reply docs/content/concepts/assets/asset-jobs.mdx:15]: Yeah, we can do that

**Review by erinkcochran87 — APPROVED** [review]

**Pull request merged**
